doc/source/java-jni/README.md

```diff
@@ -0,0 +1,237 @@
+# Packaging a Java model for Seldon Core using s2i
+
+In this guide, we illustrate the steps needed to wrap your own Java model in a
+docker image ready for deployment with Seldon Core using [source-to-image app
+s2i](https://github.com/openshift/source-to-image) and the JNI wrapper.
+
+The JNI wrapper leverages the [Python inference server](../python) to
+communicate through JNI with your Java model.
+The Python wrapper usually receives updates much faster than the [native Java
+one](../java/README.md).
+Therefore it allows you to have faster access to the latest Seldon Core
+features.
+
+If you are not familiar with s2i you can read [general instructions on using
+s2i](../wrappers/s2i.md) and then follow the steps below.
+
+## Step 1 - Install s2i
+
+[Download and install s2i](https://github.com/openshift/source-to-image#installation)
+
+ * Prerequisites for using s2i are:
+   * Docker
+   * Git (if building from a remote git repo)
+
+To check everything is working you can run
+
+```bash
+s2i usage seldonio/s2i-java-jni-build:0.3.0
+```
+
+## Step 2 - Create your source code
+
+To use our s2i builder image to package your Java model you will need:
+
+ * A Maven project that depends on `io.seldon.wrapper` library (>= `0.3.0`)
+ * A class that implements `io.seldon.wrapper.api.SeldonPredictionService` for the type of component you are creating
+ * .s2i/environment - model definitions used by the s2i builder to correctly wrap your model
+
+We will go into detail for each of these steps:
+
+### Maven Project
+Create a Spring Boot Maven project and include the dependency:
+
+```XML
+<dependency>
+	<groupId>io.seldon.wrapper</groupId>
+	<artifactId>seldon-core-wrapper</artifactId>
+	<version>0.3.0</version>
+</dependency>
+```
+
+A full example can be found at `incubating/wrappers/s2i/java-jni/test/model-template-app/pom.xml`.
+
+### Prediction Class
```

Comment:
I was going to ask if metrics and request tags are collected, but i realised they are, as metrics are collected right after predict_Raw, and tags are passed on by whatever is returned. It may be worth documenting this briefly in this section, to just mention that both are actually available

Reply:
Actually, they may not be available. We currently get metrics and tags from either the response payload or the `metrics()` method. However, we don't decode the payload at all on the Python side and we haven't hooked the `metrics()` method to the Java side yet. 

Default metrics coming from Flask should be available though.

Reply:
Interesting, ok this is good to know, may be worth making a note somewhere